src/plugins/intel_cpu/src/nodes/kernels/aarch64/brgemm_kernel.cpp

```diff
@@ -0,0 +1,342 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// Copyright (C) 2024 FUJITSU LIMITED
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "brgemm_kernel.hpp"
+#include "dnnl_extension_utils.h"
+#include "utils/cpu_utils.hpp"
+#include <cpu/aarch64/cpu_isa_traits.hpp>
+#include <openvino/core/except.hpp>
+
+using namespace dnnl::impl::cpu::aarch64;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::aarch64::matmul;
+
+#define THROW_ERROR(...) OPENVINO_THROW("brgemm executor Init Failure '", __VA_ARGS__)
+namespace ov {
+namespace intel_cpu {
+
+BrgemmKernel::BrgemmKernel(size_t M,
+                           size_t N,
+                           size_t K,
+                           size_t lda,
+                           size_t ldb,
+                           size_t ldc,
+                           bool b_transposed,
+                           ov::element::Type inType,
+                           bool b_accumulate)
+    : M(M),
+      K(K),
+      N(N),
+      lda(lda),
+      ldb(ldb),
+      ldc(ldc),
+      b_transposed(b_transposed),
+      inType(inType) {
+    // blocking M
+    bool is_bf16 = inType == ov::element::bf16;
```

Comment:
Currently, Brgemm kernel does not support f16. We are planning to support int8 brgemm kernel and once its available, I can integrate it here. For now, should I remove this check?  

For f16 path, it can go through Gemm ACL ig. I am checking the same. Thanks.